kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -210,13 +213,15 @@ class KotlinJsonAdapterFactory : JsonAdapter.Factory {
       val parameter = parametersByName[property.name]
 
       if (Modifier.isTransient(property.javaField?.modifiers ?: 0)) {
+        @Suppress("ReplaceGuardClauseWithFunctionCall") // More readable this way
         if (parameter != null && !parameter.isOptional) {
           throw IllegalArgumentException(
               "No default value for transient constructor $parameter")
         }
         continue
       }
 
+      @Suppress("ReplaceGuardClauseWithFunctionCall") // More readable this way
```

Comment:
Isn't it just `parameter == null || parameter.type == property.returnType`? That's not bad...

Reply:
Not quite - the null smartcast no longer works when this is done too. On mobile but in one of these I think it was fine, but in the other it meant nullchecking it's mention in the message